src/main/java/uk/nhs/hee/trainee/details/api/ProgrammeMembershipResource.java

```diff
@@ -71,4 +71,21 @@ public ProgrammeMembershipResource(ProgrammeMembershipService service,
         .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Trainee not found."));
     return ResponseEntity.ok(mapper.toDto(entity));
   }
+
+  /**
+   * Delete the programme memberships for the trainee.
+   *
+   * @param traineeTisId The ID of the trainee to update.
+   * @return True if the programme memberships were deleted.
+   */
+  @PatchMapping("/{traineeTisId}/delete")
```

Comment:
Seems odd to use PATCH instead of DELETE, any reason we wouldn't use `@DeleteMapping`, ?

Reply:
Yeah, I wasn't sure. I thought we are still basically patching the record (just deleting part of it, but not the entire thing). But I'm happy to take your lead on what is the most appropriate convention to use.